### PR TITLE
Rst standard format edits

### PR DESCRIPTION
**Pull request by alvizek**
Erin - There is a lot here and in hindsight I should have done these updates in batches. Take a look and then we can meet and walk through the changes together. One remaining thing that I haven't done yet is write scripts that join or transform the standard format data to be meet model needs. I can add those after this PR.

### CODE REVIEW

File: data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd [hunk 1]
```diff
@@ -423,31 +404,30 @@ unique(yuba_env$site)
...
+                          parameter == "weather" & text == "foggy" ~ "fog",
+                          parameter == "weather" & text %in% c("precipitation", "rain", "rainy", "windy and precipitation") ~ "precipitation",
+                          parameter == "weather" & text %in% c("sunny", "clear") ~ "clear",
+                          parameter == "weather" & text %in% c("windy", "windy and clear") ~ "windy",
+                          T ~ text),
+         # remove temperature above 40 because errors
+         value = case_when(parameter == "temperature" & value > 40 ~ NA_real_,
+                           T ~ value)) %>% 
+  # we do not need to keep rows for parameters where no data was collected because it does not provide any additional information
+  filter(!is.na(value) | !is.na(text)) %>%
```
[the remark this answers is not in the document]
Comment (ErinCain) [reply data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd:418]: nvm. good

File: data-raw/standard-format-data-prep/README.md [hunk 2]
```diff
@@ -20,7 +20,8 @@ This directory contains the following Rmd documents:
...
 -   [`flow_standard_format.Rmd`](https://github.com/FlowWest/JPE-datasets/blob/documentation/data-raw/standard-format-data-prep/flow_standard_format.Rmd) contains the code used to generate `standard_flow.csv` which compiles flow data from publically available streamgages.
 -   [`adult_upstream_passage_standard_format.Rmd`](https://github.com/FlowWest/JPE-datasets/blob/documentation/data-raw/standard-format-data-prep/adult_upstream_passage_standard_format.Rmd) contains the code used to generate `standard_adult_upstream_passage.csv`
--   In Progress: adult holding, redd, carcass, temperature
+-   [`holding_standard_format.Rmd`] (https://github.com/FlowWest/JPE-datasets/blob/documentation/data-raw/standard-format-data-prep/holding_standard_format.Rmd) contains the code used to generate `standard_holding.csv`
+-   In Progress: adult redd, carcass, temperature
 
 Data descriptions are included within each Rmd. Each standard dataset is described in more detail below.
```
Comment (ErinCain) [inline data-raw/standard-format-data-prep/README.md:26]: The documentation stuff all looks good. Once all the standard format things are done and we migrate the package into the JPE org we can do one more pass through of all the doc stuff

**Review by ErinCain — APPROVED** [review]

File: data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd [hunk 3]
```diff
@@ -188,41 +224,17 @@ feather_env_raw <- read_csv(here::here("data", "rst", "feather_environmental_cam
...
-  summarise(value = mean(value, na.rm = T),
-            value = ifelse(is.nan(value), NA_real_, value)) %>% 
-  rename(velocity = value)
-# convert F to C
-feather_temperature <- filter(feather_env_raw, parameter == "temperature") %>% 
-  mutate(value = case_when(unit == "°F" ~ (value-32)*(5/9),
+feather_env <- feather_env_raw %>% 
+  # convert meters to feet and get rid of nonsensical values which are errors
+  mutate(value = case_when(unit == "meters/second (m/sec)" & parameter == "velocity" ~ value*3.28,
+                           parameter == "velocity" & value > 50 ~ NA_real_,
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd:230]: checked this and the code must have been outdated. there are no longer any velocities above 50

File: data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd [hunk 4]
```diff
@@ -234,6 +246,12 @@ columns for environmental data are selected.
 
 *knights landing has 2 subsites 8.3 and 8.4. The trap environmental data that we received did not include subsite. We listed subsite as NA for these data.
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd:247]: i agree. i added comments to address this

File: data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd [hunk 5]
```diff
@@ -318,45 +309,21 @@ gcs_get_object(object_name = "rst/lower-sac-river/data-raw/tisdale/tisdale_envir
...
-  
-tisdale_env <- full_join(tisdale_flow, tisdale_velocity) %>% 
-  full_join(tisdale_temperature) %>% 
-  full_join(tisdale_turbidity) %>% 
-  mutate(stream = "sacramento river", 
-         site = "tisdale",
-         subsite = tolower(subsite))
+  mutate(subsite = tolower(subsite)) %>% 
+  # removed because large majority did not recored units
+  filter(!parameter %in% c("dissolved oxygen", "barometer", "conductivity"))
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd:326]: Looks like the unit for tisdale is likely percent. added back in with unit as percent.

File: data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd [hunk 6]
```diff
@@ -412,7 +392,8 @@ yuba_env <- read_csv(here::here("data", "rst", "yuba_rst_environmental.csv"),
          site = ifelse(subsite == "yub", "yuba river", "hallwood"),
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_environmental_standard_format.Rmd:392]: i looked at this again and think the comment about data not be associated with subsite is wrong so i removed

File: data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd [hunk 7]
```diff
@@ -139,21 +193,26 @@ butte_rst %>%
 # group by date, station, time, gear_id, trap_status then average
 butte_trap_clean <- butte_rst %>% 
   select(-c(dead, count, fork_length, weight, mark_code, lifestage,
-            weather, temperature, turbidity, secchi, velocity, gear_id)) %>%
+            weather, temperature, turbidity, secchi, velocity)) %>%
   # select(date, station, trap_status, time, gear_id, temperature, turbidity, velocity,
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd:213]: done

File: data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd [hunk 8]
```diff
@@ -551,143 +698,57 @@ combined_trap_raw <- bind_rows(battle_trap_clean,
...
-  mutate(debris_level = tolower(debris_level),
-         debris_level = ifelse(debris_level == "veryheavy", "very heavy", debris_level),
-         gear_condition = case_when(gear_condition %in% c("total block", "total blockage", "Trap not in service") ~ "trap not in service",
-                                         gear_condition %in% c("not rotating", "cone stopped", "Trap stopped functioning") ~ "trap stopped functioning",
-                                           gear_condition %in% c("partial block", "partial blockage", "Trap functioning, but not normally") ~ "trap functioning but not normally",
-                                           gear_condition %in% c("normal", "Trap functioning normally") ~ "trap functioning normally",
+  rename(trap_functioning = gear_condition_code,
+         visit_type = trap_status) %>%
+  mutate(debris_level = case_when(debris_level == "veryheavy" ~ "very heavy", 
+                                  is.na(debris_level) ~ "not recorded",
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd:747]: agreed added a line for that

File: data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd [hunk 9]
```diff
@@ -551,143 +698,57 @@ combined_trap_raw <- bind_rows(battle_trap_clean,
...
+                                  T ~ debris_level),
+         trap_functioning = case_when(trap_functioning %in% c("total block", "total blockage", "trap not in service") ~ "trap not in service",
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd:749]: fixed

File: data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd [hunk 10]
```diff
@@ -376,13 +376,18 @@ filter(feather_trap_camp, !is.na(debrisVolumeCat))
 # only a few cases where visitTime != visitTime2
 filter(feather_trap_camp, visitTime != visitTime2)
 
+# organize feather subsites by high flow channel and low flow channel to make easier to understand
+lfc <- c("eye riffle_north", "eye riffle_side channel", "gateway main 400' up river", "gateway_main1", "gateway_rootball", "gateway_rootball_river_left", "#steep riffle_rst", "steep riffle_10' ext", "steep side channel")
```
[the remark this answers is not in the document]
Comment (alvizek) [reply data-raw/standard-format-data-prep/rst_trap_standard_format.Rmd:380]: i did some checking and didn't find any cases where two different lfc/hfc sites are running at same time. most common case is herringer east and west at same time which is ok.

**Pull request merged**